### PR TITLE
Use explicit and configurable encoding when reading from files

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

**Comment by ppalaga** [comment]
@aloubyansky could you please review?

This basically adds an explicit encoding everywhere where we are reading from files. In Maven context, the encoding can be set via a new Mojo param (that defaults to `project.build.sourceEncoding` as usual in Maven). Likewise, in the CLI tool I added a new option that  defaults to `utf-8`.

**Comment by ppalaga** [comment]
@aloubyansky eb2ccb8 adds a fix for #1. I am putting it here, because it is ready for review, but I did it on top of 54c4f8b. Clearly, if you decide to reject 54c4f8b, I'll rework eb2ccb8.

File: feature-pack-api/src/main/resources/schema/pm-provisioning-1_0.xsd [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      <xs:documentation>
+        TODO: what is installation
+      </xs:documentation>
+    </xs:annotation>
+    <xs:choice maxOccurs="unbounded" minOccurs="1">
+      <xs:element name="universe" type="universe" />
+    </xs:choice>
+  </xs:complexType>
+
+  <xs:complexType name="universe">
```
Comment (aloubyansky) [inline feature-pack-api/src/main/resources/schema/pm-provisioning-1_0.xsd]: A minor note on the naming convention, I think in other schemas we follow the rule to name type as <element-name>Type. Would be good to continue it here. What do you think?

**Comment by aloubyansky** [comment]
Otherwise, it all looks good to me, thanks a lot!

File: feature-pack-api/src/main/resources/schema/pm-provisioning-1_0.xsd (hunk 1 above)
Comment (ppalaga) [inline feature-pack-api/src/main/resources/schema/pm-provisioning-1_0.xsd]: +1, let me fix it.

**Comment by ppalaga** [comment]
New or noteworthy in 3ec0690 :
- replaces eb2ccb8
- Makes the parser consistent with the XSD through making the parser reject documents without FPs or universes.
- Types in XSD now use the `*Type` naming convention.

**Comment by aloubyansky** [comment]
Thanks! Merged.

**Pull request merged**